segmentstore/contracts/src/main/java/io/pravega/segmentstore/contracts/ReadResultEntryType.java

```diff
@@ -9,30 +9,45 @@
  */
 package io.pravega.segmentstore.contracts;
 
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+
 /**
  * Defines various types of Read Result Entries, based on where their data is located.
  */
+@RequiredArgsConstructor
 public enum ReadResultEntryType {
     /**
      * The ReadResultEntry points to a location in the Cache, and data is readily available.
      */
-    Cache,
+    Cache(false),
 
     /**
      * The ReadResultEntry points to a location in Storage, and data will need to be retrieved from there in order
      * to make use of it.
      */
-    Storage,
+    Storage(false),
 
     /**
      * The ReadResultEntry points to a location beyond the end offset of the StreamSegment. It will not be able to return
      * anything until such data is appended to the StreamSegment.
      */
-    Future,
+    Future(false),
 
     /**
      * The ReadResultEntry indicates that the End of the StreamSegment has been reached. No data can be consumed
      * from it and no further reading can be done on this StreamSegment from its position.
      */
-    EndOfStreamSegment
+    EndOfStreamSegment(true),
+
+    /**
+     * The ReadResultEntry points to a location in the StreamSegment that has been truncated out and is no longer available.
+     */
+    Truncated(true);
+
+    /**
+     * If true, this indicates that the ReadResult cannot progress from this point anymore and should be closed.
```

Comment:
I'm not sure I get this comment. It says that if terminal is `true`, then `ReadResult` cannot progress. I don't understand what it means for `ReadResult` to progress. Is it referring to the reader that issue the read making progress?

I searched for `getTerminal` but couldn't find any instance of this getter.

Reply:
If your ReadResult, which is an iterator, encounters a `Truncated` ReadResultEntryType, then the ReadResult cannot iterate anymore,therefore it's a terminal entry for that ReadResult